src/components/attribution-control.js

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import {useEffect, useCallback, useState, useRef} from 'react';
+import * as PropTypes from 'prop-types';
+import mapboxgl from '../utils/mapboxgl';
+import useMapControl, {mapControlDefaultProps, mapControlPropTypes} from './use-map-control';
+
+const propTypes = Object.assign({}, mapControlPropTypes, {
+  toggleLabel: PropTypes.string,
+  className: PropTypes.string,
+  style: PropTypes.object,
+  compact: PropTypes.bool,
+  customAttribution: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)])
+});
+
+const defaultProps = Object.assign({}, mapControlDefaultProps, {
+  className: '',
+  toggleLabel: 'Toggle Attribution'
+});
+
+function setupAttributioncontrol(opts, map, container, attributionContainer) {
+  const control = new mapboxgl.AttributionControl(opts);
+  control._map = map;
+  control._container = container;
+  control._innerContainer = attributionContainer;
+
+  // from AttributionControl.onAdd()
+  control._updateAttributions();
+  control._updateEditLink();
+  map.on('styledata', control._updateData);
+  map.on('sourcedata', control._updateData);
+
+  return control;
+}
+
+function removeAttributionControl(control) {
+  control._map.off('styledata', control._updateData);
+  control._map.off('sourcedata', control._updateData);
+}
+
+function AttributionControl(props) {
+  const {context, containerRef} = useMapControl(props);
+  const innerContainerRef = useRef(null);
+  const [showCompact, setShowCompact] = useState(false);
+
+  useEffect(() => {
+    let control;
+    if (context.map) {
+      control = setupAttributioncontrol(
+        {
+          customAttribution: props.customAttribution
+        },
+        context.map,
+        containerRef.current,
+        innerContainerRef.current
+      );
+    }
+
+    return () => removeAttributionControl(control);
+  }, [context.map]);
+
+  const compact = props.compact === undefined ? context.viewport.width <= 640 : props.compact;
+
+  useEffect(() => {
+    if (!compact && showCompact) {
+      setShowCompact(false);
+    }
+  }, [compact]);
+
+  const toggleAttribution = useCallback(() => setShowCompact(value => !value), []);
+
+  return (
+    <div style={{position: 'absolute', ...props.style}} className={props.className}>
+      <div
+        ref={containerRef}
+        aria-pressed={showCompact}
+        className={`mapboxgl-ctrl mapboxgl-ctrl-attrib ${compact ? 'mapboxgl-compact' : ''} ${
```

Comment:
if we are planning on having several instances of class name manipulation we should think about using something like `classnames` to better handle these cases